docs/madr/decisions/046-policy-on-namespace.md

```diff
@@ -0,0 +1,456 @@
+the namespaced MeshTimeout is considered more specific.
+That's why the final request timeout value should be 3s.
+
+**4. Both MeshTimeout and MeshHTTPRoute in "frontend-ns"**
+
+This scenario is likely to be the most common way users create policies.
+The use case for this is:
+
+**As** a Frontend Service Owner<br>
+**I want** all outgoing requests from fronted (to backend) to have headers starting with `xxx-fronted-` removed<br>
+**so that** requests are smaller and backend doesn't receive unnecessary data.
+
+**As** a Frontend Service Owner<br>
+**I want** all requests routed to `backend` to have 3s timeout<br>
+**so that** my service meets SLO and responds withing 3s.
+
+This can be achieved by applying 2 policies:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshHTTPRoute
+metadata:
+  name: backend-route
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshService
+    name: frontend
+  to:
+    - targetRef:
+        kind: MeshService
+        name: backend
+      rules:
+        - matches:
+            - path:
+                type: PathPrefix
+                value: /
+          default:
+            filters:
+              - type: RequestHeaderModifier
+                requestHeaderModifier:
+                  remove:
+                    - xxx-frontend-version
+                    - xxx-frontend-data
+                    - xxx-frontend-content
+            backendRefs:
+              - kind: MeshService
+                name: backend
+---
+apiVersion: kuma.io/v1alpha1
+kind: MeshTimeout
+metadata:
+  name: frontend-to-backend
+  namespace: frontend-ns
+spec:
+  targetRef:
+    kind: MeshHTTPRoute
+    name: backend-route
+    namespace: frontend-ns # can be omitted as it's a default value
+  to:
+    - targetRef:
+        kind: Mesh
+      default:
+        http:
+          requestTimeout: 3s
+```
+
+**5. MeshTimeout in "frontend-ns", MeshHTTPRoute in "backend-ns"**
+
+MeshHTTPRoute in `backend-ns` produces envoy routes in DPPs in `backend-ns`.
+Ability to configure timeouts on these routes from the `frontend-ns` breaks the namespace isolation and seems
+unnecessary.
+
+**In the end,** only 3 scenarios have meaningful use cases and could be described by 2 principles:
+
+* a policy in a custom namespace can reference all other policies regardless their namespace
+* a policy in the system namespace can reference only policies in the system namespace
+
+### MeshService reference
+
+Since we didn't make any decisions on how to target MeshService in policies we will leave this out of now. We will
+revisit this while discussing targeting MeshService in policies.
+
+### MeshGateway reference
+
+`MeshGateway` at this moment is not namespace-scoped. It can change in the future with
+the [MeshBuiltinGateway resource](https://github.com/kumahq/kuma/issues/10014).
+
+We've decided that we won't allow targeting `MeshGateway` in namspace scoped polcies except for the
```

Comment:
I would say policies created in the namespace of a Service should affect requests from the MeshGateway to that service.

Reply:
You can target MeshHttpRoute from your namespace, to configure this. 

Allowing to apply policies with MeshGateway as top level targetRef could be problematic. For example, who should be able to configure policy with `to` section that targets Mesh? Example:

```yaml
apiVersion: kuma.io/v1alpha1
kind: MeshTimeout
metadata:
  name: timeout-global
  namespace: custom-ns
  labels:
    kuma.io/mesh: default
spec:
  targetRef:
    kind: MeshGateway
  to:
  - targetRef:
      kind: Mesh
```

with this you are creating policy that is modifying traffic to services from all namespaces, not only yours

Reply:
No, you'd be modifying traffic to services from only your namespace.

Reply:
To be clear, I don't think seeing `kind: Mesh` is ideal there, but that's the semantics I'm suggesting.

The point is, we have Kubernetes RBAC already and it maps very well to service ownership. It'd be a shame not to take advantage of it for policies.

Reply:
Maybe the better way would be `spec.targetRef.kind: AllMeshServices` and `from[].targetRef.kind: MeshGateway`? Either way, the use case is that as the owner `custom-ns`, I should be able to influence requests from a gateway to my services.

Reply:
I agree that it would be nice to configure these policies in your namespace, so the solution at the moment would be to disallow users setting `to: Mesh` on MeshGateway when policy is configured on custom namespace?

With `to: Mesh` allowed, there could be policies on multiple namespaces. Which policy should be applied? Should we use lexicographical order?

Reply:
Would it work to translate `kind:Mesh` into `{kind: MeshSubset, tags: {k8s.kuma.io/namespace: custom-ns}}`?

Reply:
I've updated MADR as discussed